core/includes/install.inc

```diff
@@ -741,7 +744,10 @@ make this configuration active, see admin/config/development/configuration/sync.
 ');
     }
 
-    file_put_contents($config_directory . '/README.md', $text);
+    // Only write the README.md to a file storage directory.
+    if (is_a($config_storage, 'ConfigFileStorage')) {
+      file_put_contents(config_get_config_directory($config_type) . '/README.md', $text);
+    }
```

Comment:
At this point, the `ConfigStorageInterface` derived classes have no knowledge of their `$config_type`, nor do they need it.  Adding it to support the `README.md` file in only one of them would break that abstraction.  Even the `Config` object that is the primary user of the `ConfigStorageInterface` doesn't know if the Config object itself is `active` or `staging`.

We could come up with some abstraction, like `ConfigStorageInterface::setReadme($text)` or `ConfigStorageInterface::setMetaData($key, $value)`, but the utility is still solely for `ConfigFileStorage`.  The `ConfigDatabaseStorage` class has nowhere to store a `README.md`, nor can I think of another storage class that would.  Memory? NoSQL? Memcache?